zebra/zebra_neigh.c

```diff
@@ -118,10 +118,8 @@ static void zebra_neigh_free(struct zebra_neigh_ent *n)
 		/* if rules are still using the neigh mark it as inactive and
 		 * update the dataplane
 		 */
-		if (n->flags & ZEBRA_NEIGH_ENT_ACTIVE) {
-			n->flags &= ~ZEBRA_NEIGH_ENT_ACTIVE;
-			memset(&n->mac, 0, sizeof(n->mac));
-		}
+		UNSET_FLAG(n->flags, ZEBRA_NEIGH_ENT_ACTIVE);
+		memset(&n->mac, 0, sizeof(n->mac));
```

Comment:
Earlier this memset was happening only if ZEBRA_NEIGH_ENT_ACTIVE bit was set but now it is changed to always irrespective of the flag. Looks wrong to me.

Reply:
The check simply is dumb, the flag can be unset even unconditionally, right?